network/test/unicast_authorization_test.go

```diff
@@ -408,3 +399,60 @@ func (u *UnicastAuthorizationTestSuite) TestUnicastAuthorization_PublicChannel()
 	// wait for slashing violations consumer mock to invoke run func and close ch if expected method call happens
 	unittest.RequireCloseBefore(u.T(), u.waitCh, u.channelCloseDuration, "could close ch on time")
 }
+
+// TestUnicastAuthorization_UnauthorizedUnicastOnChannel tests that messages sent via unicast that are not authorized for unicast are rejected.
+func (u *UnicastAuthorizationTestSuite) TestUnicastAuthorization_UnauthorizedUnicastOnChannel() {
```

Comment:
This test looks good but is taking a more black-box approach to assessing the unicast authorization, i.e., at the end of the test, we evaluate the receiver never receives the message which _may_ also implicitly implies an authorization failure. Please make the test more white-boxing oriented by adding a log hook and checking for the following log: 

https://github.com/onflow/flow-go/blob/796f801427998ae53cdf847fe0ea30dfac322fe8/network/p2p/middleware/middleware.go#L581-L592

Also, it would be great if we can refactor other tests in this file with the same approach.

Reply:
These tests currently do the white boxing you describe. Instead of using a log hook they use a mock slashing violations consumer https://github.com/onflow/flow-go/blob/khalil/1802-restrict-unicast-messages/network/test/unicast_authorization_test.go#L120-L133 and explicitly check that the expected Violation is called as well as the expected slashing violations consumer func is invoked. I think this is the same approach with more accurate testing than using the log hook.